### PR TITLE
Merge NDVs when computing statistics for ORs

### PR DESCRIPTION
**Pull request by vraghavan78**
Consider the OR predicate (a = 10) OR (a = 15) and the input histogram has
only NDVs. The statistics function computes result histogram for (a = 10)
as well as (a = 15) independently and then merges the two resultant
histogram in one.

The merge operation was too strict and took the maximum of the number of
distinct values (NDV)  of the two histograms to be the value for the
merged histogram. The underlying reason is correct for predicates such
as (a = 10) OR (a < 15) since this logic avoids double counting.

However, for equality operation adding up the two NDVs is fine.

It is important to note that duplicate predicates are removed during the
preprocessing step. The test added is an array comparison which is
converted into an OR before cardinality estimation

### CODE REVIEW

**Comment by sambitesh** [comment]
Can we add another easily reproducible test which leads to plan diff with and without this fix?

File: data/dxl/minidump/MinCardinalityNaryJoin.mdp [hunk 1]
```diff
@@ -13723,7 +13723,7 @@
                 <dxl:Filter/>
                 <dxl:HashJoin JoinType="Inner">
                   <dxl:Properties>
-                    <dxl:Cost StartupCost="0" TotalCost="12442.676658" Rows="5719390.578080" Width="189"/>
+                    <dxl:Cost StartupCost="0" TotalCost="22087986938243.031250" Rows="100172276179760064.000000" Width="189"/>
```
Comment (vraghavan78) [inline data/dxl/minidump/MinCardinalityNaryJoin.mdp]: this is what that makes the cost HUGE!!!

File: libnaucrates/src/statistics/CFilterStatsProcessor.cpp [hunk 2]
```diff
@@ -479,9 +481,32 @@ CFilterStatsProcessor::MakeHistHashMapDisjFilter
 			}
 			else
 			{
+
```
Comment (vraghavan78) [inline libnaucrates/src/statistics/CFilterStatsProcessor.cpp]: @bhuvnesh2703 I thought about that if then else block and felt it was more complicated to understand. May be.

**Comment by vraghavan78** [comment]
This change is causing regressions. Need to look into this. So closing it for now.

**Pull request closed without merging**